js/src/forum/components/SsoGrid.js

```diff
@@ -0,0 +1,146 @@
+import Checkbox from '../../common/components/Checkbox';
+import icon from '../../common/helpers/icon';
+import ItemList from '../../common/utils/ItemList';
+
+/**
+ * The `SsoGrid` component displays a table of linked sso providers,
+ * allowing the user to link/unlink each one to their account.
+ *
+ */
+export default class SsoGrid extends Component {
+  init() {
+    /**
+     * Information about the available Sso methods.
+     *
+     * @type {Array}
+     */
+    this.providers = this.ssoProviders().toArray();
+
+    /**
+     * A map of Sso type-method combinations to the checkbox instances
+     * that represent them.
+     *
+     * @type {Object}
+     */
+    this.inputs = {};
+
+    // For each of the Sso type-method combinations, create and store a
+    // new checkbox component instance, which we will render in the view.
+    this.providers.forEach(provider => {
+      this.inputs[provider.name] = new Checkbox({
+        state: this.providerLinked(provider.name),
+        disabled: this.disableUnlinking(provider.name),
+        onchange: () => this.toggle(provider.name),
+      });
+    });
+  }
+
+  view() {
+    return (
+      <table className="SsoGrid">
+        <thead>
+          <tr>
+            <td />
+            <th className="SsoGrid-groupToggle">
+              {app.translator.trans('core.forum.settings.sso.linked_header')}
+            </th>
+          </tr>
+        </thead>
+
+        <tbody>
+          {this.providers.map(provider => (
+            <tr>
+              <td className="SsoGrid-groupToggle">
+                {icon(provider.icon)} {provider.label}
+              </td>
+              <td className="SsoGrid-checkbox">
+                {this.inputs[provider.name].render()}
+              </td>
+            </tr>
+          ))}
+        </tbody>
+      </table>
+    );
+  }
+
+  providerLinked(provider) {
+    return app.session.user.ssoProviders().includes(provider);
+  }
+
+  disableUnlinking(provider) {
+    return this.providerLinked(provider) && !app.forum.attribute('enableUserPassAuth') && app.session.user.ssoProviders().length == 1;
+  }
+
+  /**
+   * Toggle the linked/unlinked state of the given sso provider.
+   *
+   * @param {Array} keys
+   */
+  toggle(provider) {
+    const control = this.inputs[provider];
+    control.loading = true
+
+    if (this.providerLinked(provider)) {
+      m.redraw();
+
+      app.request({
+        method: 'DELETE',
+        url: app.forum.attribute('apiUrl') + '/auth/' + provider
+      })
+        .then(() => {
+          control.props.state = false;
+          control.loading = false;
+          app.session.user.data.attributes.ssoProviders = app.session.user.ssoProviders().filter(item => item != provider);
+          for (const input in this.inputs) {
+            this.inputs[input].props.disabled = this.disableUnlinking(input);
+          }
+          m.redraw();
+        })
+        .catch(() => {
+          control.loading = false;
```

Comment:
Any error is completely silenced here I believe.

Reply:
I can confirm that this isn't the case: the red alert popup will still show up.